src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,311 @@
+    nb_components = nb_components - 1
+
+    newpixelmin = 0
+    itercount = 0
+    number_added = 20
+
+    # Maximum iteration 0 is assumed as that
+    # region growing should be carried out
+    # until all pixels are included.
+    if maxiter == 0:
+        maxiter = np.inf
+
+    # Run region growing until maximum iteration reaches
+    # and no more pixels are found
+    while (itercount < maxiter) and (number_added > newpixelmin):
+
+        # apply binary dilation
+        buffer_binary = ndimage.binary_dilation(water_binary)
+
+        # exclude the original binary pixels from buffer binary
+        buffer_binary = np.logical_xor(
+            ndimage.binary_dilation(water_binary), water_binary)
+
+        # define new_water for the pixels higher than tolerance
+        # new_water = np.logical_and(buffer_binary, defuzz > tolerance)
+        new_water = defuzz[buffer_binary] > tolerance
+
+        # add new pixels to water_binary
+        water_binary[buffer_binary] = new_water
+        number_added = np.sum(new_water)
+        itercount = itercount + 1
+        logger.info(f"iteration {itercount}: {number_added:.3f} pixels added")
+
+    return water_binary
+
+
+def process_region_growing_block(block_param,
+                                 loopind,
+                                 base_dir,
+                                 fuzzy_base_name,
+                                 input_tif_path,
+                                 initial_seed,
+                                 tolerance,
+                                 maxiter):
+
+    """Process region growing for blocks
+
+    Parameters
+    ----------
+    block_param: BlockParam
+        Object specifying where and how much to read and write to out_raster
+    loopind: integer
+        index value for iteration
+    base_dir: str
+        path for fuzzy geotiff and input geotiff
+    fuzzy_base_name: str
+        prefix for temporary file name
+    input_tif_path: str
+        path of initial fuzzy later
+    initial_seed : float
+        Threshold used for the seed of water body [0 ~ 1]
+    tolerance : float
+        relaxed threshold to be used for transient area
+        between water and non-water
+    maxiter : integer
+        maximum iteration for region growing
+
+    Returns
+    ----------
+    block_param: BlockParam
+        Object specifying where and how much to read and write to out_raster
+    data_block: numpy.ndarraya
+        fuzzy values after region growing
+    """
+
+    # At first loop, read block from intial fuzzy value geotiff
+    # Otherwise, read block from previous loop
+    if loopind == 0:
+        data_block = dswx_sar_util.get_raster_block(
+            input_tif_path, block_param)
+    else:
+        fuzzy_map_temp = \
+            f'{base_dir}/{fuzzy_base_name}_temp_loop_{loopind}.tif'
+        data_block = dswx_sar_util.get_raster_block(
+            fuzzy_map_temp, block_param)
```

Comment:
```suggestion
    if loopind == 0:
        fuzzy_map_temp = input_tif_path
    else:
        fuzzy_map_temp = \
            f'{base_dir}/{fuzzy_base_name}_temp_loop_{loopind}.tif'
    data_block = dswx_sar_util.get_raster_block(
        fuzzy_map_temp, block_param)
```
Move `get_raster_block` since it's essentially the same for both `if` and `else` and only change path passed in to `loopind == 0`